### PR TITLE
dev: switch to black as autoformatter

### PR DESCRIPTION
**Pull request by namemcguffin**
switches to black as the autoformatter

## Motivation and Context
YAPF's formatting consistently makes code less readable, especially when it comes to pipelines

## How Has This Been Tested?
tested on a private repo

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by jidicula — CHANGES_REQUESTED** [review]
instead of rolling your own Dockerfile and using 2 separate actions, why not use Black's provided action? https://github.com/psf/black#github-actions

File: README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Canary
+
  [![Discord](https://img.shields.io/discord/236668784948019202.svg)](https://discord.gg/HDHvv58)
```
Comment (jidicula) [inline README.md:3]:
```suggestion
 [![Discord](https://img.shields.io/discord/236668784948019202.svg)](https://discord.gg/HDHvv58)
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
```

**Review by jidicula — APPROVED** [review]
LGTM ⚫

File: README.md [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
+ [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
+ [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
```
Comment (davidlougheed) [inline README.md:5]: this line is duplicated?

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jidicula) [reply README.md:5]:
```suggestion
```

**Review by davidlougheed — APPROVED** [review]

**Pull request merged**